lib/sprockets/railtie.rb

```diff
@@ -125,6 +125,10 @@ def configure(&block)
       Sprockets.register_postprocessor "application/javascript", ::Sprockets::Rails::SourcemappingUrlProcessor
     end
 
+    initializer :deprecator do |app|
```

Comment:
```suggestion
    initializer "sprockets-rails.deprecator" do |app|
```

While the other initializers here don't do this (and probably never will due to backwards incompatibility), I think any new initializers added should be namespaced so that they don't conflict with any other gems.

(Duplicate initializer names has been a problem before: https://github.com/rails/rails/pull/43261#issuecomment-930336800)

Reply:
Hum there's already a namespace in the sense that all initializers are defined within a Railtie.
```
$ bin/rails initializers | grep deprecator
Sprockets::Railtie.deprecator
```
(Rails 7.0 app running with this branch)

It feels like the issue you're mentioning is more a bug/misfeature in `Railtie::Initializable` but OK.

I see all the other deprecators have a namespace in their initializer's name so that makes sense.